src/HealthGPS/dummy_model.cpp

```diff
@@ -0,0 +1,81 @@
+#include "dummy_model.h"
+
+namespace hgps {
+
+DummyModel::DummyModel(RiskFactorModelType type, const std::vector<core::Identifier> &names,
+                       const std::vector<double> &values, const std::vector<double> &policy)
+    : type_{type}, names_{names}, values_{values}, policy_{policy} {
+    if (names_.empty()) {
+        throw core::HgpsException("Risk factor names list is empty");
+    }
+    if (values_.empty()) {
+        throw core::HgpsException("Risk factor values list is empty");
+    }
+    if (policy_.empty()) {
+        throw core::HgpsException("Risk factor policy lisy empty");
+    }
+    if (names_.size() != values_.size() || names_.size() != policy_.size()) {
+        throw core::HgpsException("Risk factor name, value and policy list sizes mismatch");
+    }
+}
+
+RiskFactorModelType DummyModel::type() const noexcept { return type_; }
+
+std::string DummyModel::name() const noexcept { return "Dummy"; }
+
+void DummyModel::generate_risk_factors(RuntimeContext &context) {
+
+    // Initialise everyone.
+    for (auto &entity : context.population()) {
+        set_risk_factors(entity, context.scenario().type());
+    }
+}
+
+void DummyModel::update_risk_factors(RuntimeContext &context) {
+
+    // Initialise everyone except inactive.
+    for (auto &entity : context.population()) {
+        // Ignore if inactive.
+        if (!entity.is_active()) {
+            continue;
+        }
+
+        set_risk_factors(entity, context.scenario().type());
+    }
+}
+
+void DummyModel::set_risk_factors(Person &person, ScenarioType scenario) const {
+    for (auto i = 0u; i < names_.size(); ++i) {
+        person.risk_factors[names_[i]] = values_[i];
+
+        // Apply intervention policy.
+        if (scenario == ScenarioType::intervention) {
+            person.risk_factors[names_[i]] += policy_[i];
+        }
+    }
+}
+
+DummyModelDefinition::DummyModelDefinition(RiskFactorModelType type,
+                                           std::vector<core::Identifier> names,
+                                           std::vector<double> values, std::vector<double> policy)
+    : type_{type}, names_{std::move(names)}, values_{std::move(values)},
+      policy_{std::move(policy)} {
+    if (names_.empty()) {
+        throw core::HgpsException("Risk factor names list is empty");
+    }
+    if (values_.empty()) {
+        throw core::HgpsException("Risk factor values list is empty");
+    }
+    if (policy_.empty()) {
+        throw core::HgpsException("Risk factor policy lisy empty");
+    }
+    if (names_.size() != values_.size() || names_.size() != policy_.size()) {
+        throw core::HgpsException("Risk factor name, value and policy list sizes mismatch");
+    }
```

Comment:
This question applies equally to the other models, but why do we need to check this in both classes, the model class and the model definition class? Surely, it is impossible to have one without the other, right?

Reply:
You're right, it seems iffy. There's even an issue open about it: [#234](https://github.com/imperialCHEPI/healthgps/issues/234). Might as well just remove them now and be done with it.